sympy/core/basic.py

```diff
@@ -122,6 +122,22 @@ def __new__(cls, *args):
         obj = object.__new__(cls)
         obj._assumptions = cls.default_assumptions
         obj._mhash = None  # will be set by __hash__ method.
+        if not all(isinstance(arg, (Basic, ManagedProperties)) for arg in args ):
```

Comment:
This could be slow when there are a large number of args. I would recommend collecting unique types and then using `issubclass` like:
```python
types = set(map(type, args))
if not all(issubclass(typ, (Basic, ManagedProperties)) for typ in types):
```